packages/codemod/src/upgrade-legacy.ts

```diff
@@ -1198,43 +1198,44 @@ const upgradeLegacy = async () => {
 
   steps.push({
     name: "check for usages of invokeWithMiddleware",
-    action: async (stepIndex) => {
-      let errors = 0
-
-      getAllFiles(appDir, [], [], [".ts", ".tsx", ".js", ".jsx"]).forEach((file) => {
-        const program = getCollectionFromSource(file)
-        const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
-        if (invokeWithMiddlewarePath?.length) {
-          log.error(`\n invokeWithMiddleware found at ${file}. \n`)
-          errors++
-        }
-      })
-
+    action: async () => {
       getAllFiles(path.resolve("pages"), [], [], [".ts", ".tsx", ".js", ".jsx"]).forEach((file) => {
         const program = getCollectionFromSource(file)
-        const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
-        if (invokeWithMiddlewarePath?.length) {
-          log.error(`\n invokeWithMiddleware found at ${file}. \n`)
-          errors++
+        try {
+          const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
+          if (invokeWithMiddlewarePath?.length) {
+            invokeWithMiddlewarePath.forEach((path) => {
+              path.value.arguments.push(j.identifier("ctx"))
```

Comment:
We can also remove this?

Reply:
which part should be removed? because the loop is integral to the codemod step to work. I first get the array of all invocations of invokeWithMiddleware and then add the ctx argument to it so that when the arguments list is sliced, it will exist for the final resolver call

Reply:
> which part should be removed? because I think the loop is integral to the codemod step to work. The array of all invocations of `invokeWithMiddleware` is received and then add the `ctx` argument to it so that when the arguments list is sliced, it will exist for the final resolver call

Oops my bad. Was supposed to just highlight line `1208`. The `path.value.arguments.push(j.identifier("ctx"))`.

Reply:
@dillonraphael I am pushing the ```ctx``` into the array so that we remove the first element during

https://github.com/siddhsuresh/blitz/blob/0a8acd2e163a3e6481d759242442b2dc69200485/packages/codemod/src/upgrade-legacy.ts#L1212

```ts
path.value.arguments.slice(1)
```

The array of the ```input,ctx``` is passed into the resolver call.
Do you want be to explicitly create an array like ```[path.value.arguments[1]?.name,j.identifier("ctx")] ```

Reply:
@siddhsuresh Hmm I'm not sure I understand. I tested with the change I recommended for:

```ts
  const products = await invokeWithMiddleware(
    getProducts,
    { orderBy: { id: "desc" } },
    { req, res }
  )
 ```

and the output was:

```ts
  const products = await getProducts(
    { orderBy: { id: "desc" } },
    { req, res }
  )
```

Reply:
@dillonraphael okay, I have updated the code with your recommended change.